lib/Parse/ParseExpr.cpp

```diff
@@ -2030,7 +2032,9 @@ ParserResult<Expr> Parser::parseExprStringLiteral() {
     Status = parseStringSegments(Segments, EntireTok, InterpolationVar, 
                                  Stmts, LiteralCapacity, InterpolationCount);
 
-    auto Body = BraceStmt::create(Context, Loc, Stmts, EndLoc,
+    // At this point, PreviousLoc points to the last token parsed within
+    // the body, so use that for the brace statement location.
+    auto Body = BraceStmt::create(Context, Loc, Stmts, PreviousLoc,
```

Comment:
The source range of this `BraceStmt` is rather strange because, according to the comment, the start loc is the start of the literal and the end loc is at a token inside it, like this:
```
" blah \(some) blah \(thing) blah "
^                          ^
```
I'd recommend that it should either be the full range of the literal ([Loc, Loc]), or the range of the statements ([Stmts.first.StartLoc, Stmts.last.EndLoc]), or this range if you can manage it:
```
" blah \(some) blah \(thing) blah "
        ^                  ^
```

Reply:
I like the Stmts.{first,last} idea. I'll try that.